pkg/api/norman/customization/cluster/formatter.go

```diff
@@ -51,6 +52,21 @@ func canUserUpdateCluster(request *types.APIContext, resource *types.RawResource
 		resource.Schema) == nil
 }
 
+func canUserBackupEtcd(apiContext *types.APIContext) bool {
+	etcdBackupSchema := types.Schema{ID: mgmtclient.EtcdBackupType}
+	// pkg/rbac/access_control.go:55 canAccess checks for the object's ID or namespace. The ns for etcdbackup will be the clusterID
```

Comment:
There is no line 55 in `pkg/rbac/access_control.go:55`. This looks like it's copy-pasted from: https://github.com/rancher/rancher/blob/d3f98065499849a81f14704d7550b24126b8cd5b/pkg/api/norman/customization/cluster/actionhandler.go#L52

This comment is no longer relevant and we definitely just want to have it removed here. Also, recommend removing it from the place I've referenced while we're at it.

In fact, we're actually in the same package, so we don't need to have this copy-pasted and should just reference this newly introduced `canUserBackupEtcd` function in `actionhandler.go` and remove definition of inline `canBackupEtcd` function from `ActionHandler.ClusterActionHandler`.

Reply:
Proposal: to make the code cohesive with this change, I'd suggest moving both canUpdateCluster and canUserBackupEtcd into separate functions in ActionHandler.ClusterActionHandler. The one caveat there is that (and please correct if I'm wrong!) canUpdateCluster is called 6+ times in ActionHandler.ClusterActionHandler and it may cause less overhead as an inline func. I was thinking of having both be inline functions in ActionHandler.go and Formatter.go so the code will run faster. Thoughts?

Reply:
I don't see a need to make them have ActionHandler as a receiver as they don't need anything from the handler, so I think they should stay receiver-less.
As far as which file of `cluster` package they need to be defined in is up for debate. Maybe move them to `utils.go` as they will be used in both `actionhandler.go` and `formatter.go`?
Regarding inlining functions for performance gain, I guess the gains will be negligible (if any), so I would recommend favoring reduced copy/paste over potential trivial performance gains.

Reply:
Ok, actually. Spoke with @thedadams and since closures aren't that much faster than go functions I'm going to pull canUpdateCluster, canBackupEtcd, and canCreateClusterTemplate into shared standalone functions. I'll remove the inline definitions.

Reply:
Found this - https://stackoverflow.com/questions/45937924/in-go-do-non-capturing-closures-harm-performance
Basically, TLDR is worrying about the cost of function call when we're not talking millions of invocations is probably splitting the hairs. Plus each case is likely different as compiler optimizations may work differently based on specific circumstances. Basically, compilers are smart so we don't need to worry about minutia such as this. :)

Reply:
Sounds good! Let me know if my updates are ok

Reply:
@snasovich is good with standalone functions being in actionhandler.go. Ready for re-review